kernel/mutex.c

```diff
@@ -8,6 +8,7 @@
 #include <stdbool.h>
 #include <string.h>
 
+#include <myst/cond.h>
```

Comment:
Including cond.h here introduces a circular inclusion dependency (cond.h also includes mutex.h). This was made necessary be the definition of the `myst_thread_queue_empty()`` accessors which I would remove.

Reply:
`kernel/mutex.c` has been reverted.